lib/pscom/pscom_migrate.c

```diff
@@ -43,7 +43,7 @@ pscom_str_replace(char *search_str, char *replace_str, char *str)
 	}
 
 	/* allocate memory */
-	if ((tmp_str = (char*)malloc(strlen(str) * sizeof(char))) == NULL) {
+	if ((tmp_str = (char*)malloc(strlen(str) * sizeof(char)+ 1)) == NULL) {
```

Comment:
This shouldn't be sufficient. What if `search_str` is larger than `replace_str`? Could you update your PR such that it computes the total amount of bytes needed to hold both `str` (without `search_str`) and `replace_str`?

Reply:
I added an assert() to ensure that the sting operations do never exceed the allocated memory range.